Assets/Scripts/Balls/BallBehavior.cs

```diff
@@ -8,17 +8,74 @@ public abstract class BallBehavior
     public abstract void HandleOnCollisionEnter2D(Collision2D other);
 
     public void TransformsTo(BallType ballType) {
+        Debug.Log(this.ball.ballType+ " "+ ballType);
         Transform transform = ball.transform;
         SpriteRenderer currentSprite = ball.gameObject.GetComponent<SpriteRenderer>();
         CircleCollider2D ballCollider = ball.GetComponent<CircleCollider2D>();
 
-        if (ballType == BallType.VIRUS) {
+        if (this.ball.ballType == BallType.BALL && ballType == BallType.VIRUS) {
```

Comment:
These ball-specific transforming behavior definitions had better be put in each ball's own definition. `BallBehavior` is designed to have a common ball behavior definition regardless of its type.